src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitCyclesCheckTest.java

```diff
@@ -0,0 +1,60 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2022 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+
+import org.junit.jupiter.api.Test;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.library.freeze.FreezingArchRule;
+
+public class ArchUnitCyclesCheckTest {
+
+    /**
+     * Tests that there are no
+     * <a href="https://www.archunit.org/userguide/html/000_Index.html#_cycle_checks">cycles</a>
+     * in the package dependencies. Currently the violations from this test are frozen using
+     * <a href="https://www.archunit.org/userguide/html/000_Index.html#_freezing_arch_rules">
+     * FreezingArchRule</a>.
+     *
+     * <p>Whenever a new cycle is introduced or removed, the test will pass successfully but the
+     * frozen violations will be updated. It is highly recommended to avoid new cycles and, it
```

Comment:
Question: I have not been part of the conversation but CI will fail on unstaged changes after run, right?

Reply:
Yes